scripts/update-references-in-vsce.js

```diff
@@ -0,0 +1,214 @@
+          delete packagingDeps[oldDep];
+          updated = true;
+        }
+        // Update bundling script
+        if (json.scripts && json.scripts["bundle:extension"]) {
+          let bundlingScript = json.scripts["bundle:extension"];
+          const newDep = referencesMap[oldDep];
+          if (bundlingScript.includes(oldDep)) {
+            bundlingScript = bundlingScript.replace(new RegExp(`--external:${oldDep}`, 'g'), `--external:${newDep}`);
+            updated = true;
+          }
+          json.scripts["bundle:extension"] = bundlingScript;
+        }
+      }
+
+      // If updated, write the modified package.json back to disk
+      if (updated) {
+        fs.writeFile(filePath, JSON.stringify(json, null, 2), (err) => {
+          if (err) {
+            console.error(`Error writing file ${filePath}: ${err}`);
+          } else {
+            console.log(`Updated dependencies in ${filePath}`);
+          }
+        });
+      }
+    });
+  }
+
+  // Function to update package.json directly under the packages directory
+  function traverseDirectories(directory) {
+    fs.readdir(directory, { withFileTypes: true }, (err, files) => {
+      if (err) {
+        console.error(`Error reading directory ${directory}: ${err}`);
+        return;
+      }
+
+      files.forEach(file => {
+        if (file.isDirectory()) {
+          const packageJsonPath = path.join(directory, file.name, 'package.json');
+          if (fs.existsSync(packageJsonPath)) {
+            updateDependencies(packageJsonPath);
+          }
+        }
+      });
+    });
+  }
+
+  // Start traversing from the packages directory
+  traverseDirectories(packagesDir);
+}
+
+function updateImports() {
+  const dirs = ['src', 'test'];
+  function replaceTextInFile(filePath) {
+    const data = fs.readFileSync(filePath, 'utf8');
+    let result = data.replace(
+      /'@salesforce\/core'/g,
+      "'@salesforce/core-bundle'"
+    );
+    result = result.replace(
+      /'@salesforce\/core\/(.+)'/g,
+      "'@salesforce/core-bundle'"
+    );
+    result = result.replace(
+      /'@salesforce\/source-deploy-retrieve/g,
+      "'@salesforce/source-deploy-retrieve-bundle"
+    );
+    result = result.replace(
+      /'@salesforce\/source-tracking/g,
+      "'@salesforce/source-tracking-bundle"
+    );
+    result = result.replace(
+      /'@salesforce\/templates/g,
+      "'@salesforce/templates-bundle"
+    );
+    result = result.replace(
+      /'@salesforce\/apex-node/g,
+      "'@salesforce/apex-node-bundle"
+    );
+
+    fs.writeFileSync(filePath, result, 'utf8');
+  }
+  function traverseDirectory(directory) {
+    fs.readdirSync(directory).forEach((file) => {
+      const fullPath = path.join(directory, file);
+      if (fs.lstatSync(fullPath).isDirectory()) {
+        traverseDirectory(fullPath);
+      } else if (path.extname(fullPath) === '.ts') {
+        replaceTextInFile(fullPath);
+      }
+    });
+  }
+
+  function traversePackages(directory) {
```

Comment:
What's special about the `packages` directory compared to a regular directory that it needs a separate traverse function?  Why can't we just call `traverseDirectory(packagesDir);` in Line 154 and get rid of this function?  It seems like that's what you are already doing in `updateEsbuildConfig()`, right?  (That being said, if `traverseDirectories()` is going to be moved to the top level, can that be shared in `updateImports()` too?)